### PR TITLE
field255: Implement `TryFrom<Field255>` for `u64`

### PR DESCRIPTION
**Pull request by cjpatton**
Based on #431 (merge that first).
Partially addresses #141.

Poplar1 unsharding requires conversion from the leaf field to `u64`.

### CODE REVIEW

File: src/field/field255.rs [hunk 1]
```diff
@@ -289,6 +289,24 @@ impl Default for Field255 {
...
 
+impl TryFrom<Field255> for u64 {
+    type Error = FieldError;
+
+    fn try_from(elem: Field255) -> Result<u64, FieldError> {
+        let mut le_bytes = [0; 32];
+        fiat_25519_to_bytes(&mut le_bytes, &elem.0);
+        let mut is_zero = Choice::from(1);
+        for byte in &le_bytes[8..] {
+            is_zero &= !Choice::from(*byte);
```
Comment (divergentdave) [inline src/field/field255.rs:300]: This constructor should only be used with `0u8` or `1u8`, changing the "too_big" test to use `[2; 32]` hits a debug assertion inside subtle. Instead, we should use `<u8 as ConstantTimeEq>` methods.

**Review by divergentdave — CHANGES_REQUESTED** [review]

File: src/field/field255.rs [hunk 2]
```diff
@@ -289,6 +289,24 @@ impl Default for Field255 {
...
 }
 
+impl TryFrom<Field255> for u64 {
+    type Error = FieldError;
+
+    fn try_from(elem: Field255) -> Result<u64, FieldError> {
+        let mut le_bytes = [0; 32];
+        fiat_25519_to_bytes(&mut le_bytes, &elem.0);
+        let mut is_zero = Choice::from(1);
+        for byte in &le_bytes[8..] {
```
Comment (tgeoghegan) [inline src/field/field255.rs:299]: nit: to make it more clear why we're checking that the last 24 bytes are 0, `use std::mem::size_of` and:
```suggestion
        for byte in &le_bytes[size_of::<u64>()..] {
```

**Review by tgeoghegan — CHANGES_REQUESTED** [review]

**Comment by cjpatton** [comment]
Rebased.

**Review by divergentdave — APPROVED** [review]

**Review by tgeoghegan — APPROVED** [review]

**Comment by cjpatton** [comment]
Rebased once more!

**Pull request merged**